libraries/node-core-library/src/Terminal/Colors.ts

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+/**
+ * @beta
+ */
+export interface IColorableSequence {
+  text: string;
+  foregroundColor?: ColorValue;
+  backgroundColor?: ColorValue;
+}
+
+/**
+ * @beta
+ */
+export enum ColorValue {
+  Black,
+  Red,
+  Green,
+  Yellow,
+  Blue,
+  Magenta,
+  Cyan,
+  White,
+  Gray
+}
+
+/**
+ * @beta
+ */
+export class Colors {
+  public static black(text: string | IColorableSequence): IColorableSequence {
+    return {
+      ...(typeof text === 'string' ? { text } : text),
+      foregroundColor: ColorValue.Black
+    };
+  }
+
+  public static red(text: string | IColorableSequence): IColorableSequence {
+    return {
+      ...(typeof text === 'string' ? { text } : text),
+      foregroundColor: ColorValue.Red
+    };
+  }
+
+  public static green(text: string | IColorableSequence): IColorableSequence {
+    return {
+      ...(typeof text === 'string' ? { text } : text),
+      foregroundColor: ColorValue.Green
+    };
+  }
+
+  public static yellow(text: string | IColorableSequence): IColorableSequence {
+    return {
+      ...(typeof text === 'string' ? { text } : text),
+      foregroundColor: ColorValue.Yellow
+    };
```

Comment:
Could you put an if-then statement in here?  My head is exploding trying to understand this.